superbench/benchmarks/micro_benchmarks/directx_gemm_flops_performance/Options.h

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <sstream>
+#include <string>
+
+namespace Option {
+enum Precision {
+    F16,
+    F32,
+};
+using PrecisionType = Option::Precision;
+} // namespace Option
+
+struct Options {
+    char **begin;
+    char **end;
+
+    /**
+     * @brief Get the char* value of the cmd line argument.
+     * @param option the argument in cmd.
+     * @return char*
+     */
+    char *get_cmd_option(const std::string &option) {
+        char **itr = std::find(begin, end, option);
+        if (itr != end && ++itr != end) {
+            return *itr;
+        }
+        return 0;
+    }
+
+    /**
+     * @brief Get the int type value of cmd line argument.
+     * @param option the cmd line argument.
+     * @return int the int type value of cmd line argument 'option'.
+     */
+    int get_cmd_line_argument_int(const std::string &option, int defaults) {
+        if (char *value = get_cmd_option(option)) {
+            return std::stoi(value);
+        }
+        return defaults;
+    }
+
+    /**
+     * @brief Get the string type value of cmd line argument.
+     * @param  option the cmd line argument.
+     * @return std::string the int type value of cmd line argument 'option'.
+     */
+    std::string get_cmd_line_argument_string(const std::string &option) {
+        if (char *value = get_cmd_option(option)) {
+            return std::string(value);
+        }
+        return "";
+    }
+
+    /**
+     * @brief Get the boolean type value of cmd line argument.
+     * @param  option the cmd line argument.
+     * @return bool the boolean value.
+     */
+    bool get_cmd_line_argument_bool(const std::string &option) {
+        if (char *value = get_cmd_option(option)) {
+            bool b;
+            std::istringstream(std::string(value)) >> b;
+            return b;
+        }
```

Comment:
actually I am not quite understanding this code. I guess you don't need to following a number for this parameter. If the parameter "--fp16" exists, it should be set to true. For now you need to set "--fp16 1" to set this flag, and I don't think it is natural way.

Reply:
so how about the code like the following:
if (cmd_option_exists(option))
{
      return true;
}

return false;

so your command line like "-m 100 -n 100 --fp16"

Reply:
ok, updated to 
```
    bool get_cmd_line_argument_bool(const std::string &option) {
        if (cmd_option_exists(option)) {
            return true;
        }
        return false;
    }
```